Documentation for Just in time mapping (#492)
precice/precice.github.io: pages/docs/configuration/configuration-mapping.md

The mapping configuration has undergone a major revision between preCICE version 2 and preCICE version 3. For the documentation of version 2, checkout the documentation of [our previous versions](fundamentals-previous-versions.html).
{% endnote %}

{% tip %}
Instead of using static meshes as described here, we are currently working on a [just-in-time data mapping](couple-your-code-just-in-time-mapping.html), where the meshes are not statically defined during initialization, but change at runtime.

> {% endtip  %}

Reviewer: Could we move this tip further down? Maybe even the note above as well. Let's start with the main content first and explain alternatives later.
And please make the docs timeless. "we are currently working on" will be outdated eventually and we will forget to update it.

Author: Moved it in front of the 'Available methods' section and made it timeless.

For a sequential participant, any combination of `read`/`write`-`consistent/conservative` is valid. For a participant running in parallel, only `read`-`consistent` and `write`-`conservative` is possible. More details are given [further below](configuration-mapping.html#restrictions-for-parallel-participants).

{% tip %}

> Instead of using static meshes as described here, you may consider using a [just-in-time data mapping](couple-your-code-just-in-time-mapping.html), where the meshes are not statically defined during initialization, but change at runtime.

Reviewer: ```suggestion
Instead of using static meshes as described here, you may consider using a [just-in-time data mapping](couple-your-code-just-in-time-mapping.html), where one of both meshes can change at runtime.
```
